Sources/DB/DTGItemRealm.swift

```diff
@@ -15,21 +15,27 @@ import RealmSwift
 class TrackInfoRealm: Object {
     @objc dynamic var title: String = ""
     @objc dynamic var languageCode: String = ""
+    @objc dynamic var type: String = ""
+    @objc dynamic var selected = false
     
-    convenience init(trackInfo: TrackInfo) {
+    convenience init(itemId: String, selected: Bool, trackInfo: TrackInfo) {
         self.init()
         self.title = trackInfo.title
         self.languageCode = trackInfo.languageCode
+        self.type = trackInfo.type.rawValue
+        self.selected = selected
+        
     }
     
+    
     public override class func shouldIncludeInDefaultSchema() -> Bool { return false } 
 
     func asTrackInfo() -> TrackInfo {
-        return TrackInfo(languageCode: self.languageCode, title: self.title)
+        return TrackInfo(languageCode: self.languageCode, title: self.title, type: TrackInfo.TrackType(rawValue: type)!)
```

Comment:
Let's solve this in a different way, without '!'.
Don't want it to crash by any chance.